src/dfx/src/commands/start.rs

```diff
@@ -320,6 +325,26 @@ pub fn exec(
                 }
             }
 
+            let previous_config_path = replica_config_dir.join("replica-effective-config.json");
```

Comment:
Let's store this in the network data directory, not the replica configuration directory.  Two reasons:
1. The replica configuration directory isn't especially useful, as it only stores the port and pid, but not actually the replica configuration.  We should probably instead store those two files in the network data directory
2. We need this "require clean" functionality on more than just the replica configuration, specifically to also include: replica/emulator mode, and replica commit hash